datafusion/Cargo.toml

```diff
@@ -81,6 +81,7 @@ num-traits = { version = "0.2", optional = true }
 pyo3 = { version = "0.15", optional = true }
 tempfile = "3"
 parking_lot = "0.12"
+croaring = "0.5.1"
```

Comment:
I am somewhat worried about adding this dependency to datafusion (at least as a required dependency). Not only will it increase the requirements / compile time for datafusion, it also complicates the build process (as now one needs to install clang to build datafusion)

Reply:
@alamb Thanks for your reply. I agree It 's complicated for add a new dependency `clang` into datafusion.
IMO, we still need an high-efficiency bitmap implement `roarling-bitmap`, like IOx and other high-end OLAP systems to do some optimize like late materialization. Maybe we should change `croaring ` to `roarling-rs` (completely written in rust, won't cost too mush compile time, i also find datafusion compile\build parallel ), maybe one day it will catch up with `croaring ` 😂

Reply:
I agree with @alamb , would be good if we can either make this dependency optional or change to a pure rust implementation.